### ISSUE TITLE
More message search filtering

### ISSUE THREAD

**Issue by tezlm** [ajbura/cinny#1421]
### Describe the problem

The current search feature is pretty nice, but lacks some quality of life features.

### Describe the solution you'd like

- filter search results by user (yes, [this is possible](https://spec.matrix.org/v1.8/client-server-api/#post_matrixclientv3search) (`filter.senders: [...]`), but no client seems to implement it yet?)
- search all rooms
- search all rooms in a space (including sub-spaces)

### Additional context

These feature requests are all in one issue since they are all very related and seem to be ui + various permutations of `/search` filters, but I can split it into multiple issues if necessary.

**Comment by ajbura** [comment]
- [ ] filter search results by user (yes, [this is possible](https://spec.matrix.org/v1.8/client-server-api/#post_matrixclientv3search) (filter.senders: [...]), but no client seems to implement it yet?)
- [X] search all rooms
- [X] search all rooms in a space (including sub-spaces)

**Comment by filips123** [comment]
I agree that more search options would be useful. For example, Discord offers various useful [search filters](https://support.discord.com/hc/en-us/articles/115000468588-How-to-Use-Search-on-Discord):

- User
- Mentions
- Type (file, image, video, link, etc.)
- Date (before date, after date, during date)
- Channel/room
- Is pinned
- Author type (user, bot, webhook)

All filters can be combined together.

<img width="455" height="510" alt="Discord search options" src="https://github.com/user-attachments/assets/9dbbbab3-7ea1-439f-b9aa-42523c991e53" />

Example of why search by user and date might be useful: You know that someone sent some message that wou want to find, but you don't know what exactly it contained, so you can't search for it directly. However, you know who sent it, in which room, and roughly when. So, you can search for all messages by that user in a specific room during some month, so you can find that message more easily.
